api/clients/eigenda_client_v2.go

```diff
@@ -0,0 +1,142 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"github.com/Layr-Labs/eigenda/api/clients/codecs"
+	corev2 "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/hashicorp/go-multierror"
+	"math/rand"
+)
+
+// EigenDAClientV2 provides the ability to get blobs from the relay subsystem, and to send new blobs to the disperser.
+type EigenDAClientV2 interface {
+	// GetBlob iteratively attempts to retrieve a given blob with key blobKey from the relays listed in the blobCertificate.
+	GetBlob(ctx context.Context, blobKey corev2.BlobKey, blobCertificate corev2.BlobCertificate) ([]byte, error)
+	GetCodec() codecs.BlobCodec
+	Close() error
+}
+
+type eigenDAClientV2 struct {
+	clientConfig EigenDAClientConfig
+	log          logging.Logger
+	relayClient  RelayClient
+	codec        codecs.BlobCodec
+}
+
+var _ EigenDAClientV2 = &eigenDAClientV2{}
+
+// BuildEigenDAClientV2 builds an EigenDAClientV2 from config structs.
+func BuildEigenDAClientV2(
+	log logging.Logger,
+	clientConfig EigenDAClientConfig,
+	relayClientConfig RelayClientConfig) (EigenDAClientV2, error) {
+
+	err := clientConfig.CheckAndSetDefaults()
+	if err != nil {
+		return nil, err
+	}
+
+	relayClient, err := NewRelayClient(&relayClientConfig, log)
```

Comment:
if you're taking a reference here, any reason not to take a pointer to RelayClientConfig as argument to this function?

Reply:
No reason, will fix

Reply:
Done